cloud/aws/elasticache/common/monitors-elasticache.tf

```
@@ -0,0 +1,174 @@
+	}
+
+}
+
+resource "signalfx_detector" "no_connection" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] ElastiCache connections"
+
+	program_text = <<-EOF
+		signal = data('CurrConnections', filter=filter('namespace', 'AWS/ElastiCache') and filter('stat', 'lower') filter('CacheNodeId', '*') and ${module.filter-tags.filter_custom})${var.no_connection_aggregation_function}.${var.no_connection_transformation_function}(over='${var.no_connection_transformation_window}')
+		detect(when(signal <= ${var.no_connection_threshold_critical})).publish('CRIT')
+	EOF
+
+	rule {
+		description           = "are too low <= ${var.no_connection_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.no_connection_disabled_critical, var.no_connection_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.no_connection_notifications_critical, var.no_connection_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "swap" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] ElastiCache swap"
+
+	program_text = <<-EOF
+		signal = data('SwapUsage', filter=filter('namespace', 'AWS/ElastiCache') and filter('stat', 'upper') and filter('CacheNodeId', '*') and ${module.filter-tags.filter_custom})${var.swap_aggregation_function}.${var.swap_transformation_function}(over='${var.swap_transformation_window}')
+		detect(when(signal > ${var.swap_threshold_critical})).publish('CRIT')
+		detect(when(signal > ${var.swap_threshold_warning})).publish('WARN')
+	EOF
+
+	rule {
+		description           = "is too high > ${var.swap_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.swap_disabled_critical, var.swap_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.swap_notifications_critical, var.swap_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "is too high > ${var.swap_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.swap_disabled_warning, var.swap_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.swap_notifications_warning, var.swap_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "free_memory" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] ElastiCache free memory"
```

```suggestion
 	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] AWS ElastiCache free memory"
```
idealy, use a 100 - free memory to use > instead of < logic